### PR TITLE
Invoice payment processor

### PR DESCRIPTION
**Pull request by mjfrey**
Payment processor that records an order for Coupons and the Client who this is for.

This allows an order to be processed without collection of funds during the order process.

### CODE REVIEW

**Comment by mjfrey** [comment]
@vkaracic @marjev please review.

File: ecommerce/extensions/api/serializers.py [hunk 1]
```diff
@@ -145,6 +146,13 @@ class Meta(object):
         model = Refund
 
 
+class InvoiceSerializer(serializers.ModelSerializer):
+    """ Serializer for Invoice objects. """
+
+    class Meta(object):
+        model = Invoice
+
```
Comment (vkaracic) [inline ecommerce/extensions/api/serializers.py]: Remove whitespace here.

File: ecommerce/extensions/api/v2/tests/views/test_invoices.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+from ecommerce.invoice.models import Invoice
+from ecommerce.tests.testcases import TestCase
+
+Client = get_model('core', 'Client')
+
+DEFAULT_DATE_FORMAT = '%Y-%m-%dT%H:%M:%S.%fZ'
+
+
+class InvoiceViewTests(APITestCase, TestCase):
```
Comment (vkaracic) [inline ecommerce/extensions/api/v2/tests/views/test_invoices.py]: You don't need to extend APITestCase.

File: ecommerce/extensions/api/v2/tests/views/test_invoices.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
+from __future__ import unicode_literals
```
Comment (vkaracic) [inline ecommerce/extensions/api/v2/tests/views/test_invoices.py]: You import ``unicode_literals`` which means you don't need to prefix your strings with ``u' '``.

File: ecommerce/invoice/models.py [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+"""
+Model to capture that an invoice was generated for a given order.
+"""
+
+
+class Invoice(models.Model):
+    invoice_number = models.CharField(max_length=20)
+    order_number = models.CharField(max_length=20)
+    client = models.ForeignKey('core.Client')
+    state = models.CharField(max_length=10, default='Not Paid', choices=[('Not Paid', 'Not Paid'), ('Paid', 'Paid')])
```
Comment (vkaracic) [inline ecommerce/invoice/models.py]: Not sure if it's important, but usually the choices are created as constants like this:
https://github.com/django-oscar/django-oscar/blob/master/src/oscar/apps/partner/abstract_models.py#L238

File: ecommerce/extensions/api/v2/views/invoices.py [hunk 5]
```diff
@@ -0,0 +1,15 @@
+"""HTTP endpoints for interacting with invoices."""
+from rest_framework.permissions import IsAuthenticated, IsAdminUser
+from rest_framework import generics
+
+from ecommerce.invoice.models import Invoice
+from ecommerce.extensions.api import serializers
```
Comment (marjev) [inline ecommerce/extensions/api/v2/views/invoices.py]: Imports need to be alphabetized.

**Comment by mjfrey** [comment]
@vkaracic @marjev please check I think I addressed all the comments.

**Comment by mjfrey** [comment]
@dsego can you take a look?

**Comment by vkaracic** [comment]
:+1:

**Comment by marjev** [comment]
:+1:

**Comment by dsego** [comment]
Looks good.

**Pull request closed without merging**